src/Domain.jl

```diff
@@ -1577,6 +1577,29 @@ end
 end
 export calcdomainderivatives!
 
+function jacobiany!(jac::Q,y::U,p::W,t::Z,domain::D,interfaces::Q3,colorvec::Q2=nothing) where {Q3<:AbstractArray,Q2,Q<:AbstractArray,U<:AbstractArray,W,Z<:Real,D<:ConstantTPDomain}
+    ns,cs,T,P,V,C,N,mu,kfs,krevs,Hs,Us,Gs,diffs,Cvave = calcthermo(domain,y,t,p)
+    jacobiany_ns!(jac,y,p,t,domain,[],cs,kfs,krevs,nothing)
+    for i in domain.indexes[1]:domain.indexes[2]
+        jac[domain.indexes[3],i] = sum(jac[domain.indexes[1]:domain.indexes[2],i])*R*T/P
```

Comment:
an @views on the jac slice might help avoid allocating a new vector corresponding to jac[domain.indexes[1]:domain.indexes[2],i] each iteratioon.

Reply:
Thanks for the suggestion! Added!